### ISSUE TITLE
The indent complains about correctly indented expressions with parentheses

### ISSUE THREAD

**Issue by w0rp** [eslint/eslint#8666]
**Tell us about your environment**

* **ESLint Version:** v4.0.0-beta.0
* **Node Version:** v6.4.0
* **npm Version:** yarn v0.21.3

**What parser (default, Babel-ESLint, etc.) are you using?**

The default parser.

**Please show your full configuration:**

```
// .eslintrc.js
module.exports = {
  root: true,
  extends: "eslint:recommended",
  rules: {indent: ["warn", 2]},
}
```

**What did you do? Please include the actual source code causing the issue.**

Consider the following code, which uses parentheses to group some math which is up-indented
at the opening parenthesis. `eslint` says that the expression with the parentheses is not correctly indented.

```js
(
  1
  + (
    2  // eslint says 2 more spaces are needed here...
    - 3
  )  // ... and here.
)
```

If you apply `--fix`, you will get the following:

```js
(
  1
  + (
      2
    - 3
    )
)
```

If you write this, `eslint` doesn't complain at all:

```js
(
  1
  + 
  (
    2
    - 3
  )
)
```

**What did you expect to happen?**

These results don't make sense. The expressions inside the parentheses should be indented with only one extra level of indentation, not 2. You should be able to add a closing parenthesis on the same level of indentation as the level you opened it on. The second line which starts with an operator seemed to be ignored.

This bug doesn't exist in the alpha version, so it seems to have been introduced between the alpha and beta versions.

**What actually happened? Please include the actual, raw output from ESLint.**

`eslint` complained about what appears to be valid indentation.

**Comment by not-an-aardvark** [comment]
Thanks, I can confirm this.

However, I was confused by this comment:

> This bug doesn't exist in the alpha version, so it seems to have been introduced between the alpha and beta versions.

As far as I can tell, the behavior has been the same since at least 4.0.0-alpha.0. Are you sure this worked in the alpha version?

**Comment by w0rp** [comment]
I do see the bug here in the alpha version also, so I'm not sure. I first noticed this when updating from the alpha to the beta in a codebase, which caused a number of warnings to appear.

**Comment by not-an-aardvark** [comment]
Would you mind giving an example of the other warnings that appeared when you upgraded from alpha to beta? It's possible there is a second bug that caused those warnings to appear in the beta, and it would be nice to fix that too.

**Comment by w0rp** [comment]
I'll have a look through the other warnings when I can. I believe it was mostly the same issue a number of times throughout the codebase. The beta did also fix the indentation in one file better than the alpha, as the indentation level was too high for one entire file.

**Comment by Pyrolistical** [comment]
Hmm, this looks related to https://github.com/eslint/eslint/issues/8717
